backend/agent_loop.py

```diff
@@ -133,11 +154,18 @@ def system_message(self) -> str:
         return system_message
 
 
-def create_agent(llm, tools, workspace_path):
+def create_agent(llm, tools, workspace_path, use_remote_runtime: bool = False):
     """Create an agent with development tools and workspace configuration"""
-    # Create a custom agent context to store workspace_path for template rendering
-    agent_context = CustomAgentContext(workspace_path=workspace_path)
+    # For remote runtimes, tell the agent about the remote workspace path
+    if use_remote_runtime:
+        agent_workspace_path = "/workspace"
+    else:
+        agent_workspace_path = workspace_path
 
+    # Create custom agent context with workspace path
+    agent_context = CustomAgentContext(workspace_path=agent_workspace_path)
+
+    # Use CustomAgent to get proper system message with workspace_path template
     return CustomAgent(llm=llm, tools=tools, agent_context=agent_context)
```

Comment:
@openhands get rid of the CustomAgent class entirely. We should ONLY use the built-in Agent class in the SDK.

Get rid of CustomAgentContext as well. We should ONLY use the built-in AgentContext.

Build Agent and AgentContext objects using their constructors to pass in things like system prompts